Header footer resume check in

asorichetti opened the pull request:
Think I may have an issue as my footer is shooting back the error that "'footerRef.current' is possibly 'null'."

asorichetti committed:
Getting there on the restart

asorichetti committed:
Mid step

asorichetti committed:
Fixed changes at least ot an extent

asorichetti committed:
Made it so there is a background color
todo: Make background color apply to entire page
not just lines with content

asorichetti committed:
Added footer and resume styles as well as adding
resume and fixing multiple components

asorichetti committed:
Finished writting footer code - currently has
error due to the fact footerRef.current can be
"null"

hweeks commented on src/components/Footer/index.tsx:
```diff
@@ -0,0 +1,40 @@
+                window.removeEventListener('scroll', handleScrollios);
+            }
+        };
+    });
```
is this still needed in modern iOS? it is rougnly 4 year old code

asorichetti replied:
Went with this after viewing my own mobile device browser, will look into documentation to be sure that it is still a necessity

hweeks commented on src/components/Footer/index.tsx:
```diff
@@ -0,0 +1,40 @@
+        <Footer>
+            <FootText ref={footerRef}>
+                <FootLink href="www.linkedin.com/in/alex-sorichetti">LinkedIn</FootLink>
+                -
```
this should be encoded as an html entity

asorichetti replied:
FootLink is an <a> assigned variable, but I do not currently know if I need to tweak this any farther for it to be working.

asorichetti replied:
Added in opening "&" and closing ";" around text to encode as HTML elements

hweeks commented on src/components/Footer/index.tsx:
```diff
@@ -0,0 +1,40 @@
+                <FootLink href="www.linkedin.com/in/alex-sorichetti">LinkedIn</FootLink>
+                -
+                <ResumeHome></ResumeHome>
+                -
```
same here

hweeks commented on src/components/Footer/styles.ts:
```diff
@@ -0,0 +1,23 @@
+    text-align: center;
+    font-size: 3rem;
+    -webkit-text-fill-color: #F5D547;
+    `;
```
can you make the spacing and line ends consistent? 9 and 15/23 are different. also a new line at the bottom of a file is standard, you can make your editor do this automaitcally

asorichetti replied:
Fixed with the new line as well as consistent tab indents

hweeks commented on src/components/Footer/styles.ts:
```diff
@@ -0,0 +1,23 @@
+    font-size: 3rem;
+    -webkit-text-fill-color: #F5D547;
+    @media (max-width: 768px){
+        font-size: 12vw;
```
is there a reason to lock font size to viewport width?

asorichetti replied:
Changed this to rem rather than vw as there was no valid reason for this.

hweeks commented on src/components/resume/index.tsx:
```diff
@@ -0,0 +1,23 @@
+    const toggleOpen = () => setOpen(!open);
+    return(
+        <StyledLayout>
+            <IfOpenTrue open={!open}>
```
if there is nothing in a tag, you can self end it

```
<IfOpenTrue open={!open} />
```

asorichetti replied:
Will fix up to this point

asorichetti replied:
Fixed formatting here

hweeks commented on src/components/resume/index.tsx:
```diff
@@ -0,0 +1,23 @@
+            </IfOpenTrue>
+        
+        <IfOpenTrue open={open}>
+        <div>
```
why the extra div around the div you return?

hweeks commented on src/components/resume/styles.ts:
```diff
@@ -0,0 +1,12 @@
+    text-decoration: underline;
+    text-decoration: overline;
+    cursor: pointer
+    `;
```
can we fix spacing and newline here?

asorichetti replied:
Corrected indents and added new line to the bottom

hweeks commented on src/index.tsx:
```diff
@@ -1,10 +1,40 @@
-import React from "react";
+import * as React from "react";
```
this style of import should ne be needed and indicates following an older react guideline. why was this changed?

asorichetti replied:
Similar to the resume import, this is a auto-complete and auto-correction suggestion that went wrong, has since been deleted

hweeks commented on src/index.tsx:
```diff
@@ -1,10 +1,40 @@
-    Home
+
+return( 
+  <div style={styles.container}>
```
please stick to styled-components, mixing two types of styling makes it hard to reason about how stlying is handled

asorichetti replied:
styles.container is referencing the const that is described which sets the background fill, as well as setting text padding, and colour. Haven't yet changed this part, but I will double-check styled styled-components for if there is a built-in edit for the background colour and making sure the back fill goes edge to edge.

hweeks commented on static/index-template.html:
```diff
@@ -7,6 +7,7 @@
   <meta name="viewport" content="width=device-width, initial-scale=1.0" />
   <title>
     <%= htmlWebpackPlugin.options.title %>
+    <link href="https://fonts.googleapis.com/css2?family=Montserrat&display=swap" rel="stylesheet">
```
a link is not a valid child of the title, it must be in the head

asorichetti replied:
removed from the title and placed within the head

hweeks commented on tsconfig.json:
```diff
@@ -11,5 +11,5 @@
     "media.d.ts",
     "src/**/*"
-  ]
+, "src/components/resume"  ]
```
why have you added a single file? this is handeld by the `src/**/*` import

asorichetti replied:
I didn't recall adding this, but it has been removed after your note

hweeks reviewed (changes requested):
a series of styling and approach updates are needed

asorichetti committed:
Fixed issue with spacing and formatting as well
as removing unneccisary call in tsconfig.json to
resume file that had already been being called.
Fixed issue where text was locked to view width
rather than simply refrencing it as an rem unit

asorichetti committed:
Fixed further issue formatting issue with spacing
after the final line of code segments.

asorichetti committed:
Fixed external link formatting as well as fixing
remaining indentation and moved font style
out of title and into the head in index-template

asorichetti committed:
Fixed href links to reflect use of styled.a in the
FootLink reference in style

asorichetti committed:
Fixed formatting in src/components/resume/index
for ease of reading with proper indentation.

asorichetti committed:
Added HTML entity surrounding string text

asorichetti committed:
Undid change of &; surroundings on things that
did not require it.

asorichetti committed:
Fixed HTML Encoding via rule of escaped entities.
